web/server/codechecker_server/api/report_server.py

```diff
@@ -3279,3 +3279,124 @@ def getAnalysisStatistics(self, run_id, run_history_id):
                                               failedFilePaths=failed_files,
                                               successful=stat.successful)
         return analyzer_statistics
+
+    @exc_to_thrift_reqfail
+    @timeit
+    def exportData(self, run_filter):
+        self.__require_access()
+
+        with DBSession(self.__Session) as session:
+
+            run_ids = run_filter.ids
+
+            # Logic for getting comments
+            comment_data_list = defaultdict(list)
+            comment_query = session.query(Comment, Report.bug_id) \
+                .filter(Report.run_id.in_(run_ids)) \
+                .outerjoin(Report, Report.bug_id == Comment.bug_hash) \
+                .order_by(Comment.created_at.desc())
+
+            for data, report_id in comment_query:
+                comment_data = comment_data_db_to_api(data)
```

Comment:
Yes, I agree. Although thought it would be better for the user to understand the comments in the outputted json file if we format it. But yes, that would require extra processing on both importer and exporter side